service.go

```diff
@@ -168,14 +168,17 @@ func (s *service) Run() error {
 		return err
 	}
 
-	ch := make(chan os.Signal, 1)
-	signal.Notify(ch, syscall.SIGTERM, syscall.SIGINT, syscall.SIGQUIT)
-
-	select {
-	// wait on kill signal
-	case <-ch:
-	// wait on context cancel
-	case <-s.opts.Context.Done():
+	if s.opts.DisableSignalHandler {
+		<-s.opts.Context.Done()
```

Comment:
Surely if `if s.opts.Signal {` then we add `signal.Notify(ch, syscall.SIGTERM, syscall.SIGINT, syscall.SIGQUIT)`. In this way the code will remain clean as before, but we'll just never receive anything on the `ch` channel.

Reply:
this is literally what I was doing in #943. Which one do you prefer?

Reply:
Your comments: https://github.com/micro/go-micro/pull/943#issuecomment-554111145 am I confused?

Reply:
Misunderstanding

```
ch := make(chan os.Signal)

if s.opts.Signal {
    signal.Notify(ch, syscall.SIGTERM, syscall.SIGINT, syscall.SIGQUIT)
}

// rest remains the same
select {
case <-ch:

case <-s.opts.Context.Done():

}
```